src/app/components/ManualEntry.svelte

```diff
@@ -84,50 +132,87 @@
     font: 1em var(--main-font);
   }
 
-  input:focus {
+  table input:focus {
     background-color: var(--lite-white);
   }
+
+  table p {
+    display: inline;
+    color: var(--red);
+  }
+
+  .save-zone {
+    display: flex;
+    flex-direction: row;
+    align-items: center;
+    justify-content: center;
+    padding: 1%;
+  }
 </style>
 
 <div class="language__sources-manual-entry">
   <div class="language__sources-manual-entry-tablename">
     <h4>Table Name</h4>
-    <input data-cy="manual-entry-input-tablename" />
+    <input
+      type="text"
+      bind:value={tableData.name}
+      data-cy="manual-entry-input-tablename" />
   </div>
 
-  <table>
+  <table id="manual-entry-table">
     <thead>
-      <th>Word</th>
-      <th>Count</th>
-      <th>Action</th>
+      {#each columns as column (column.name)}
+        <th>{column.label}</th>
+      {/each}
+      <th />
     </thead>
 
+    {#each rowDataFromManualEntry as row, i (i)}
+      <tr>
+        <td>
+          <input
+            type="text"
+            bind:value={row.word}
+            data-cy="manual-entry-input-word" />
+          {#if !validateRowData(row)}
+            <p>*</p>
+          {/if}
```

Comment:
Some of the validation can be done automatically by the browser. If you want to make sure the field is filled, use [the `required` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/required). No JavaScript needed!

```suggestion
          <input
            type="text"
            bind:value={row.word}
            required
            data-cy="manual-entry-input-word" />
```

I would not keep the \*, but if you really want it, wrap it in a `<span>`; not a `<p>`